src/lib/checkpoint/cp_mgr.cpp

```diff
@@ -69,24 +71,29 @@ void CPManager::register_consumer(cp_consumer_t consumer_id, std::unique_ptr< CP
     }
 }
 
+[[nodiscard]] CPGuard CPManager::cp_guard() { return CPGuard{}; }
+
 CP* CPManager::cp_io_enter() {
     rcu_read_lock();
     auto cp = get_cur_cp();
     if (!cp) {
         rcu_read_unlock();
         return nullptr;
     }
+    cp_ref(cp);
+    rcu_read_unlock();
+
+    return cp;
+}
 
-    auto cnt = cp->m_enter_cnt.fetch_add(1);
+void CPManager::cp_ref(CP* cp) {
+    cp->m_enter_cnt.fetch_add(1);
```

Comment:
Should this have `memory_order_relaxed` ordering since it's not protecting anything (that I see)?

Reply:
This was original code as written in 1.2/1.3, I didn't modify it. However, I don't think it can be `memory_order_relaxed` because the enter_cnt once become zero, we need a barrier. This is same pattern we have in many part of our code and we generalized in sisl::atomic_counter (which uses acq/release pairs). I will replace that with atomic_counter, so we have one place to modify.